finetune/lora.py

```diff
@@ -0,0 +1,225 @@
+"""
+Instruction-tuning with LoRA on the Alpaca dataset.
+
+Note: If you run into a CUDA error "Expected is_sm80 to be true, but got false", uncomment the line
+`torch.backends.cuda.enable_flash_sdp(False)` in the script below (see https://github.com/Lightning-AI/lit-llama/issues/101).
+"""
+import sys
+from typing import Literal
+from pathlib import Path
+import os
+import time
+
+import lightning as L
+import numpy as np
+import torch
+
+# support running without installing as a package
+wd = Path(__file__).parent.parent.resolve()
+sys.path.append(str(wd))
+
+from generate.base import generate
+from lit_parrot.lora import mark_only_lora_as_trainable, lora, lora_state_dict
+from lit_parrot.model import Parrot, Config
+from lit_parrot.tokenizer import Tokenizer
+from lit_parrot.utils import lazy_load, check_valid_checkpoint_dir # TODO: Use lazy_load
+from scripts.prepare_alpaca import generate_prompt
+
+
+instruction_tuning = True
+eval_interval = 100
+save_interval = 100
+eval_iters = 100
+log_interval = 1
+
+# Hyperparameters
+learning_rate = 3e-4
+batch_size = 128
+micro_batch_size = 4
+gradient_accumulation_iters = batch_size // micro_batch_size
+assert gradient_accumulation_iters > 0
+max_iters = 50000 * 3 // micro_batch_size
+weight_decay = 0.0
+max_seq_length = 256  # see scripts/prepare_alpaca.py
+lora_r = 8
+lora_alpha = 16
+lora_dropout = 0.05
+warmup_iters = 100
+
+
+def main(
+    data_dir: Path = Path("data/alpaca"),
+    checkpoint_dir: Path = Path("checkpoints/stabilityai/stablelm-base-alpha-3b"),
+    out_dir: Path = Path("out/lora/alpaca"),
+    precision: Literal["bf16-true", "32-true", "bf16-mixed"] = "bf16-true",
+):
+
+    print(checkpoint_dir)
```

Comment:
Lefover print? The "loading model" print will include this alreay

Reply:
Yes, thanks!